control-plane-operator/controllers/hostedcontrolplane/oauth/config.go

```diff
@@ -52,9 +52,11 @@ func ReconcileOAuthServerConfig(ctx context.Context, cm *corev1.ConfigMap, owner
 
 func generateOAuthConfig(ctx context.Context, client crclient.Client, namespace string, params *OAuthConfigParams) (*osinv1.OsinServerConfig, error) {
 	var identityProviders []osinv1.IdentityProvider
-	identityProviders, _, err := convertIdentityProviders(ctx, params.IdentityProviders, params.OauthConfigOverrides, client, namespace)
+	identityProviders, _, err := ConvertIdentityProviders(ctx, params.IdentityProviders, params.OauthConfigOverrides, client, namespace)
 	if err != nil {
-		return nil, err
+		// Eat the error here, since we don't want to fail the deployment if the identity providers are invalid
```

Comment:
It would be nice if we also had a trace of this error in the logs when debugging, could we log it as a warning ?

Reply:
This would be nice but we don't have access to the `logger` for the `HostedControlPlaneReconciler` here and piping in new arguments will likely created backporting pains.

Reply:
I logged it where we eval the condition so that it still appears in the CPO logs :+1: